include/engine/routing_algorithms/shortest_path.hpp

```diff
@@ -114,7 +114,11 @@ class ShortestPathRouting final
                           needs_loop_forwad,
                           needs_loop_backwards);
         }
-        new_total_distance += std::min(total_distance_to_forward, total_distance_to_reverse);
+        //if no route is found between two parts of the via-route, the entire route becomes invalid
+        if( new_total_distance == INVALID_EDGE_WEIGHT )
```

Comment:
Ah okay, but the line below then `new_total_distance = INVALID_EDGE_WEIGHT;` is basically just a noop?

It might be easier to read if we just check `if( new_total_distance != INVALID_EDGE_WEIGHT )` and then execute the current else branch?